File: types/tournament.go
package main

import "time"

type Permission struct {
	ID    int64
	Type  string
	Flags int
}

type City struct {
	ID    int64
	Name  string
	State string
}

type Account struct {
	ID            int64
	Email         string
	Password      string
	CreatedOn     time.Time
	LoggedOn      time.Time
	Verified      bool
	Token         int64
	PermissionsID int64
}

type AccountPermission struct {
	PermissionsID int64
	AccountsID    int64
}

type Team struct {
	ID          int64
	CitiesID    int64
	Name        string
	Description string
}

type Player struct {
	ID         int64
	AccountsID int64
	TeamsID    int64
	FirstName  string
	LastName   string
	FullName   string
	Birthday   time.Time
	Number     int
	Height     int
	Weight     int
	Wingspan   int
	Position   string
}

type TeamPlayer struct {
	TeamsID   int64
	PlayersID int64
}

type Audit struct {
	ID      int64
	Time    time.Time
	Status  int
	Message string
}

File: types/tournament.ts
interface Permission {
    id: number;
    type: 'player' | 'referee' | 'admin';
    flags: number;
}

interface City {
    id: number;
    name: string;
    state: string;
}

interface Account {
    id: number;
    email: string;
    password: string;
    created_on: Date;
    logged_on: Date;
    verified: boolean;
    token: number;
    permissions_id: number;
}

interface AccountPermission {
    permissions_id: number;
    accounts_id: number;
}

interface Team {
    id: number;
    cities_id: number | null;
    name: string;
    description: string;
}

interface Player {
    id: number;
    accounts_id: number;
    teams_id: number | null;
    first_name: string;
    last_name: string;
    full_name: string;
    birthday: Date;
    number: number;
    height: number;
    weight: number;
    wingspan: number;
    position: 'PG' | 'SG' | 'SF' | 'PF' | 'C';
}

interface TeamPlayer {
    teams_id: number;
    players_id: number;
}

interface Audit {
    id: number;
    time: Date;
    status: number;
    message: string;
}